pkg/maincmd/maincmd.go

```diff
@@ -29,7 +30,14 @@ type readWriter struct {
 func (r *readWriter) Read(p []byte) (n int, err error)  { return r.r.Read(p) }
 func (r *readWriter) Write(p []byte) (n int, err error) { return r.w.Write(p) }
 
-func Main(args []string, stdin io.Reader, stdout io.Writer, stderr io.Writer, cfg *config.Config) error {
+func Main(
+	ctx context.Context,
+	args []string,
+	stdin io.Reader,
+	stdout io.Writer,
+	stderr io.Writer,
+	cfg *config.Config,
+) error {
```

Comment:
My editor showed me it was becoming a long line and so I split it, but I later realized there are even longer lines for func declarations, so I can revert this to be in line with the project style.

Even with the context it was still not so long as to conflict with human rights :)

Reply:
Project style is to not split argument lists, so yes, please revert this change. I prefer to have people’s editor handle line wrapping (within reason).

Too long function signatures are also a sign that maybe we should use an option struct for this function :)